q2cli/util.py

```diff
@@ -73,6 +77,43 @@ def exit_with_error(e, header='An error has been encountered:',
     click.get_current_context().exit(status)
 
 
+def output_in_cache(fp):
+    """Determines if an output path follows the format
+    /path_to_extant_cache:key
+    """
+    from pathlib import Path
+    from qiime2.core.cache import Cache
+
+    # Tells us right away this isn't in a cache
+    if ':' not in fp:
+        return False
+
+    split_path = fp.split(':')
+
+    # Account for potential for : in the path
+    cache_path = ':'.join(elem for elem in split_path[:-1])
+    key = split_path[-1]
+
+    try:
+        if Cache.is_cache(Path(cache_path)):
```

Comment:
We should probably make `Cache.is_cache` indifferent to the input type. Accepting either `str` or `Path`

Reply:
Turns out I already am doing that. I already caste it to a pathlib.Path at the beginning of is_cache.